### PR TITLE
Add hint when snapshot fails due to large file

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: cli/src/cli_util.rs [hunk 1]
```diff
@@ -206,7 +206,16 @@ impl From<OpHeadResolutionError<CommandError>> for CommandError {
 
 impl From<SnapshotError> for CommandError {
     fn from(err: SnapshotError) -> Self {
-        CommandError::InternalError(format!("Failed to snapshot the working copy: {err}"))
+        match err {
+            SnapshotError::NewFileTooLarge { .. } => user_error_with_hint(
+                format!("Failed to snapshot the working copy: {err}"),
+                "Increase the value of the `snapshot.max-new-file-size` config option if you \
+                 \nwant this file to be snapshotted. Otherwise add it to your `.gitignore` file.",
```
Comment (PhilipMetzger) [inline cli/src/cli_util.rs:213]: Use a r#"" string for the second part.

**Review by PhilipMetzger — APPROVED** [review]
LG

File: cli/src/cli_util.rs (hunk 1 above)
Comment (martinvonz) [reply cli/src/cli_util.rs]: Done. Also got rid of a trailing space.

File: cli/src/commands/mod.rs [hunk 2]
```diff
@@ -3594,7 +3594,7 @@ fn cmd_workspace_update_stale(
         workspace_command.unchecked_start_working_copy_mutation()?;
     match check_stale_working_copy(&locked_wc, &desired_wc_commit, &repo) {
         Ok(_) => {
-            locked_wc.discard();
+            // locked_wc.discard();
```
Comment (PhilipMetzger) [inline cli/src/commands/mod.rs:3597]: Remove this commented out line?

Comment (martinvonz) [reply cli/src/commands/mod.rs]: Oops, done.

**Pull request merged**